src/controllers/UserController.js

```diff
@@ -0,0 +1,37 @@
+const asyncHandler = require("express-async-handler");
+const User = require("../database/UserModel");
+
+// @desc   Adds new user if it doesnt exist and returns newly created user, otherwise return existing user from database
```

Comment:
nit - you want a tiny bit more info here. what defines a unique user? by username? by email? etc

Reply:
added description defining a unique user